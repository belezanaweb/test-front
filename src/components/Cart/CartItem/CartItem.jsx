import React from 'react'

import jss from 'jss'
import preset from 'jss-preset-default'

jss.setup(preset())

const styles = {
  card: {
    display: 'flex',
    padding: [12, 5],
    borderRadius: 3,
    margin: { bottom: 15 },
    border: {
      width: 1,
      style: 'solid',
      color: '#EEE',
    },
    '&:last-child': {
      margin: { bottom: 0 },
    }
  },
  column: {
    padding: [0, 5],
    '&:last-child': {
      flex: 'auto',
    }
  },
  thumb: {
    height: 65,
    width: 65,
  },
  productName: {
    display: 'block',
    lineHeight: '16px',
    fontSize: 13,
    color: '#212122',
    height: 32,
    overflow: 'hidden',
    margin: { bottom: 15 },
    textAlign: 'left',
  },
  productPrice: {
    display: 'block',
    color: '#212122',
    lineHeight: '17px',
    fontSize: 14,
    fontWeight: 'bold',
    textAlign: 'right',
  },
}

const { classes } = jss.createStyleSheet(styles).attach()

const CartItem = ({
  name,
  thumb,
  price,
}) =>
  <div className={classes.card}>
    <div className={classes.column + ' ' + classes.thumb}>
      <img src={thumb} title={name} className={classes.thumb} />
    </div>
    <div className={classes.column}>
      <span className={classes.productName}>{name}</span>
      <span className={classes.productPrice}>R$ {price.toFixed(2).replace('.', ',')}</span>
    </div>
  </div>

export default CartItem
