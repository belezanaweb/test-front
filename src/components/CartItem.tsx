import PriceTag from './PriceTag'
import ProductPicture from './ProductPicture'

export default function CartItem({
  product,
  showPrice = true
}: {
  product: Product,
  showPrice?: boolean
}) {
  return (
    <div className="grid grid-cols-12 gap-3.5 justify-between content-center items-center mb-9 last:mb-0">
      <div className="col-span-2 h-[60px] min-w-[60px]">
        <ProductPicture imageObject={product.imageObjects[0]} alt={product.name} />
      </div>
      <div className="col-span-7 overflow-hidden max-h-[50px] pl-4">
        <p className="text-xs leading-4 text-clip ">{product.name}</p>
      </div>
      {showPrice && (
        <div className="col-span-3 flex flex-col">
          <PriceTag priceSpecification={product.priceSpecification} />
        </div>
      )}
    </div>
  )
}
