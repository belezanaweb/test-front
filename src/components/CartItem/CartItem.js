import React from 'react'
import { createUseStyles } from 'react-jss'

const useStyles = createUseStyles({
  cartItem: {
    display: 'flex',
    alignItems: 'center',
    justifyContent: 'space-between'
  },
  cardItemText: {
    fontFamily: 'Helvetica Neue',
    fontSize: '13px',
    letterSpacing: '0.93px',
    lineHeight: '16px'
  },
  cardItemPrice: {
    fontFamily: 'Helvetica Neue',
    fontSize: '13px',
    letterSpacing: '0.93px',
    fontWeight: 'bold',
    textAlign: 'right',
    width: '100%'
  },
  withWrapper: {
    display: 'flex',
    alignItems: 'flex-start',
    border: 'solid #EEE 1px',
    borderRadius: '3px',
    boxSizing: 'border-box',
    flexWrap: 'nowrap',
    flexDirection: 'row',
    marginBottom: '15px',
    justifyContent: 'space-between',
    padding: '0 10px'
  },
  productInfo: {
    flex: '1'
  }
})

const CartItem = (props) => {
  const { imageSrc, name, price } = props
  const classes = useStyles()

  return (
    <div className={(classes.cartItem, classes.withWrapper)}>
      <div>
        <img src={imageSrc} alt={name}></img>
      </div>
      <div className={classes.productInfo}>
        <p className={classes.cardItemText}>{name}</p>
        <p className={classes.cardItemPrice}>{price}</p>
      </div>
    </div>
  )
}

export default CartItem
