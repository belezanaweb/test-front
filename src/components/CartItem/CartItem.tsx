import PriceTag from '../PriceTag'
import ProductPicture from '../ProductPicture'

export default function CartItem({
  product,
  showPrice = true
}: {
  product: Product,
  showPrice?: boolean
}) {
  return (
    <div className="grid grid-cols-12 gap-3.5 md:gap-2 justify-between content-center items-center mb-9 last:mb-6 first:mt-3">
      <div className="col-span-2 h-[60px] md:h-full min-w-[60px] md:min-w-[90px] md:col-span-2">
        <ProductPicture imageObject={product.imageObjects[0]} alt={product.name} />
      </div>
      <div
        className={`${
          showPrice ? 'col-span-7' : 'col-span-10'
        } overflow-hidden max-h-[50px] md:max-h-fit pl-4 md:pl-6 md:mt-7`}
      >
        <p className="text-xs md:text-sm leading-4 text-clip ">{product.name}</p>
      </div>
      {showPrice && (
        <div className="col-span-3 flex flex-col  md:mt-7">
          <PriceTag priceSpecification={product.priceSpecification} />
        </div>
      )}
    </div>
  )
}
