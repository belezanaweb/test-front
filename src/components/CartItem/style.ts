import styled from 'styled-components'

export const SectionContainer = styled.section`
  width: 100%;
  margin-bottom: 5px;
  @media (min-width: 1024px) {
    width: 100%;
  }
`

export const Container = styled.section`
  width: 100%;
  padding: 13px;
  background-color: #fff;
  border-radius: 3px;
  margin-bottom: 20px;
  box-shadow: 2px 2px 5px rgba(0, 0, 0, 0.2);
`

export const Title = styled.h4`
  font-size: 0.875rem;
  font-weight: 800;
  color: #999;
  text-transform: uppercase;
  padding: 0 10px;
  margin-bottom: 5px;
`
