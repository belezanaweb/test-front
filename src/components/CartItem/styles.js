import styled from 'styled-components'

export const CartItemStyles = styled.li`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 10px 0;
  list-style: none;
  border: 1px solid #ddd;
  border-radius: 5px;
  padding: 10px;
  .product-details {
    margin-left: 10px;
  }
  .product-name {
    font-size: 0.8rem;
    font-weight: 400;
    line-height: 1.4;
    margin-top: 0;
    margin-bottom: 6px;
  }
  .product-price {
    font-size: 0.84rem;
    font-weight: bold;
    text-align: right;
    display: block;
    margin-bottom: 0px;
  }
`
