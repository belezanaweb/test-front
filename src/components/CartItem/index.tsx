import CartItem from './CartItem'

export default CartItem