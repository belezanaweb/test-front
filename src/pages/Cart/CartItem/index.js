import React from 'react'

import { Container, ImageContainer, DescriptionContainer, Name, Price } from './styles'

import useWindowDimensions from '../../../utils/windowDimension'
import convertValue from '../../../utils/convertValue'

function CartItem({ item }) {
  const { width } = useWindowDimensions()

  const { product } = item
  const { imageObjects: image, name, priceSpecification } = product
  const { originalPrice } = priceSpecification
  return (
    <Container>
      <ImageContainer>
        {width <= 768 ? (
          <img alt={name} src={image[0].thumbnail}></img>
        ) : width <= 1024 ? (
          <img alt={name} src={image[0].small}></img>
        ) : width <= 1440 ? (
          <img alt={name} src={image[0].medium}></img>
        ) : (
          <img alt={name} src={image[0].large}></img>
        )}
      </ImageContainer>
      <DescriptionContainer>
        <Name>{name}</Name>
        <Price>{convertValue(originalPrice)}</Price>
      </DescriptionContainer>
    </Container>
  )
}

export default CartItem
