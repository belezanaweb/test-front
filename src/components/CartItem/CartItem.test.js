import React from 'react'
import renderer from 'react-test-renderer'
import CartItem from './CartItem'

it('renders correctly', () => {
  const tree = renderer.create(<CartItem></CartItem>).toJSON()
  expect(tree).toMatchSnapshot()
})
