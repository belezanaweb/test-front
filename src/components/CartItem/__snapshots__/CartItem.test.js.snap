// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="withWrapper-0-2-4"
>
  <div>
    <img />
  </div>
  <div
    className="productInfo-0-2-5"
  >
    <p
      className="cardItemText-0-2-2"
    />
    <p
      className="cardItemPrice-0-2-3"
    />
  </div>
</div>
`;
